ingestors/manager.py

```diff
@@ -148,14 +193,34 @@ def ingest(self, file_path, entity, **kwargs):
         entity.set("processingAgent", get_distribution("ingest").version)
         entity.set("processedAt", now_string)
 
+        ingestor_class = None
+        ingestor_name = None
+
         try:
             ingestor_class = self.auction(file_path, entity)
-            log.info("Ingestor [%r]: %s", entity, ingestor_class.__name__)
+            ingestor_name = ingestor_class.__name__
+            log.info("Ingestor [%r]: %s", entity, ingestor_name)
+
+            start_time = default_timer()
             self.delegate(ingestor_class, file_path, entity)
+            duration = max(0, default_timer() - start_time)
+
+            INGESTIONS_SUCCEEDED.labels(ingestor=ingestor_name).inc()
+            INGESTION_DURATION.labels(ingestor=ingestor_name).observe(duration)
+
+            if file_size is not None:
+                INGESTED_BYTES.labels(ingestor=ingestor_name).inc(file_size)
+
             entity.set("processingStatus", self.STATUS_SUCCESS)
         except ProcessingException as pexc:
-            entity.set("processingError", stringify(pexc))
             log.exception("[%r] Failed to process: %s", entity, pexc)
+
+            if ingestor_name:
+                INGESTIONS_FAILED.labels(ingestor=ingestor_name).inc()
```

Comment:
Done! Actually I think we don’t even need the `or None`, because `ingestor_name` is initialized as `None` a few lines before. So it will be either overwritten, or it will still be `None`.